Makefile

```diff
@@ -8,96 +8,57 @@ endif
-
-build-app-test:
-		docker build . --build-arg aws_secret_access_key=${AWS_SECRET_ACCESS_KEY} \
-		--build-arg aws_access_key_id=${AWS_ACCESS_KEY_ID} \
-		--build-arg okta_client_id=${OKTA_CLIENT_ID} \
-		--build-arg okta_client_secret=${OKTA_CLIENT_SECRET} \
-		--build-arg REG=${REG} \
-		-t ${REG}/agr_literature_app_test:${TAG} \
-		-f ./docker/Dockerfile.app-test.env
-
+build:
+	docker-compose build
 
 run-flake8:
-	docker run --rm -v ${PWD}:/workdir -i ${REG}/agr_literature_dev:${TAG} /bin/bash -c "python3 -m flake8 ."
+	docker-compose --env-file .env.test down -v
+	docker-compose --env-file .env.test run -v ${PWD}:/workdir test_runner /bin/bash -c "python3 -m flake8 ."
+	docker-compose --env-file .env.test down -v
 
 run-local-flake8:
 	python3 -m flake8 .
 
 run-mypy:
-	docker run --rm -v ${PWD}:/workdir -i ${REG}/agr_literature_dev:${TAG} /bin/bash -c "mypy --config-file mypy.config ."
+	docker-compose --env-file .env.test down -v
+	docker-compose --env-file .env.test run -v ${PWD}:/workdir test_runner /bin/bash -c "mypy --config-file mypy.config ."
+	docker-compose --env-file .env.test down -v
 
 run-local-mypy:
 	mypy --config-file mypy.config .
 
 run-dev-bash:
-	docker run --rm \
-	    --network=main-app_agr-literature \
-	    -p ${API_PORT}:8080 \
-	    -v ${PWD}:/workdir \
-		-t -i ${REG}/agr_literature_dev:${TAG} \
-		/bin/bash
+	docker-compose --env-file .env.test run --service-ports --rm dev_app /bin/bash
 
 run-dev-zsh:
-	docker run --rm -v "${HOME}/.vimrc:/root/.vimrc:rw" -v "${HOME}/.zshrc:/root/.zshrc:rw" -v ${PWD}:/workdir -t -i ${REG}/agr_literature_dev:${TAG} /bin/zsh
+	docker-compose --env-file .env.test run --service-ports --rm -v "${HOME}/.vimrc:/root/.vimrc:rw" -v "${HOME}/.zshrc:/root/.zshrc:rw" -v ${PWD}:/workdir dev_app /bin/zsh
 
 docker-compose-up:
 	docker run -itd --env-file=.env -v /var/run/docker.sock:/var/run/docker.sock -v /home/core/.docker:/root/.docker -v ${PWD}:/var/tmp/ docker/compose:1.24.1  -f /var/tmp/docker-compose.yaml up -d
 
 docker-compose-down:
 	docker run -itd --env-file=.env -v /var/run/docker.sock:/var/run/docker.sock -v /home/core/.docker:/root/.docker -v ${PWD}:/var/tmp/ docker/compose:1.24.1  -f /var/tmp/docker-compose.yaml down 
 
-run-test-bash: build-env build-dev
-	-docker volume rm -f agr-literature-test_agr-literature-pg-data
+run-test-bash:
+	docker-compose --env-file .env.test down -v
 	docker-compose --env-file .env.test up -d postgres
 	sleep 5
-	# Minus at start means ignore exit code for that line
-	-docker run -i --rm \
-		--network=agr-literature-test_agr-literature \
-	    -p 8080:8080 \
-		-e PYTHONPATH:/workdir/src/xml_processing/ \
-	    -v ${PWD}:/workdir \
-		${REG}/agr_literature_dev:${TAG} \
-		./run_tests.sh > pytest.out
-	docker-compose --env-file .env.test down
-    #doing here after shutdown of database 
+	-docker-compose --env-file .env.test run -v ${PWD}:/workdir test_runner ./run_tests.sh > pytest.out
```

Comment:
There's a leading - here, but maybe that's supposed to be there ?

Reply:
the leading - suppresses errors and it's needed here.

Reply:
Good to know, thanks !